jans-linux-setup/jans_setup/setup_app/utils/base.py

```
@@ -381,20 +382,48 @@ def mylog(logs):
     opener.addheaders = headers
     urllib.request.install_opener(opener)
 
-    mylog("Downloading {} to {}".format(url, dst))
+    mylog(f"Downloading {url} to {dst}")
     download_tries = 1
+    download_ok = False
     while download_tries < 4:
         try:
-            urllib.request.urlretrieve(url, dst)
+            dst_tmp_fn = dst + '~' + os.urandom(4).hex()
+            urllib.request.urlretrieve(url, dst_tmp_fn)
```

Ruff (0.15.21)

[error] Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)